haskell-src/Concordium/ID/Types.hs

```diff
@@ -271,10 +274,45 @@ instance ToJSONKey IdentityProviderIdentity where
 -- Account signatures (eddsa key)
 type AccountSignature = Signature
 
--- encryption key for accounts (Elgamal?)
+-- |An Elgamal encryption key. Note that this is a full key with a specific
+-- generator @g@ and the @g^s@ for a secret key @s@. In our use the generator is
+-- in fact always the same (it is part of the shared cryptographic parameters of
+-- the chain, set in genesis), so we could put it into a shared context instead
+-- to save some space.
 newtype AccountEncryptionKey = AccountEncryptionKey {_elgamalPublicKey :: ElgamalPublicKey}
     deriving (Eq, Show, Serialize, FromJSON, ToJSON) via ElgamalPublicKey
 
+-- |A marker type that has the 'FBS.FixedLength' instance to determine the size
+-- of the account encryption key in bytes.
+data AccountEncryptionKeySize
+
+instance FBS.FixedLength AccountEncryptionKeySize where
+  fixedLength _ = 2 * 48
+
+-- |A byte representation of the account encryption key. This is both smaller
+-- (1/3 the size) of the 'AccountEncryptionKey' as well as much faster to load
+-- since it does not involve checking whether the key is a valid point on the
+-- relevant elliptic curve. As a result, this can only be used when the input is
+-- either trusted to be well-formed, or when checks will be done on it before
+-- the value is needed as an encryption key.
+newtype RawAccountEncryptionKey = RawAccountEncryptionKey (FBS.FixedByteString AccountEncryptionKeySize)
+    deriving(Eq, Ord)
+    deriving (Show, Serialize, FromJSON, ToJSON) via FBSHex AccountEncryptionKeySize
+
+-- |Reconstruct the account encryption key from the raw representation. This
+-- function is unsafe and will raise an exception if the input is not a valid
+-- account encryption key.
+unsafeEncryptionKeyFromRaw :: RawAccountEncryptionKey -> AccountEncryptionKey
+unsafeEncryptionKeyFromRaw (RawAccountEncryptionKey fbs) =
+  case decode (FBS.toByteString fbs) of
+    Left _ -> error "Precondition violation. Invalid encryption key."
```

Comment:
It's not clear from this error message what to do to fix the underlying problem.

Perhaps, at this time this error message should at least be prepended with a caveat "not supposed to happen", so whoever sees it after the node crashes knows that it's not them who did something wrong but us.